plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestingIgniteServer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import io.trino.testing.ResourcePresence;
+import io.trino.testing.SharedResource;
+import io.trino.testng.services.ManageTestResources;
+import io.trino.util.AutoCloseableCloser;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+
+import static io.trino.plugin.ignite.TestIgniteContainer.HTTP_PORT;
+import static java.lang.String.format;
+
+@ManageTestResources.Suppress(because = "The actual server is kept on a static field and instance of this class is used for reference counting only")
+public class TestingIgniteServer
+        implements AutoCloseable
+{
+    private static final String IGNITE_IMAGE = "apacheignite/ignite:2.8.0";
+
+    @GuardedBy("this")
+    private static SharedResource<TestingIgniteServer> sharedResource;
+    private static AutoCloseableCloser autoCloseableCloser;
+
+    private final TestIgniteContainer dockerContainer;
+
+    public static synchronized TestingIgniteServer getInstance()
```

Comment:
Why not to return the `Lease` here so the user will take care of freeing (decrementing the counter) this.

Reply:
Done